imagery-cesium/plugins/entities-visualizer/base-entities-visualizer.ts

```diff
@@ -1,112 +1,245 @@
-					),
-					pixelOffset: Cesium.Cartesian2.ZERO,
-					eyeOffset: Cesium.Cartesian3.ZERO,
-					heightReference: Cesium.HeightReference.NONE,
-					horizontalOrigin: Cesium.HorizontalOrigin.CENTER,
-					verticalOrigin: Cesium.VerticalOrigin.CENTER,
-					scale: 1.0,
-					image: entity.icon,
-					imageSubRegion: undefined,
-					color: Cesium.Color.WHITE,
-					id: id,
-					rotation: 0.0,
-					alignedAxis: Cesium.Cartesian3.ZERO,
-					width: undefined,
-					height: undefined,
-					scaleByDistance: undefined,
-					translucencyByDistance: undefined,
-					pixelOffsetScaleByDistance: undefined,
-					sizeInMeters: false,
-					distanceDisplayCondition: undefined
-				};
-				const feature = this.billboardCollection.add(billboardEntity);
-				this.idToEntity.set(id, <any>{
-					originalEntity: entity,
-					feature: feature
-				});
+			const featureJson: Feature<any> = entity.featureJson;
+
+			const newEntities: Entity[] = [];
+
+			switch (featureJson.geometry.type) {
+				case 'Point': {
+					if (entity.icon) {
+						newEntities.push(this.getBillboard(entity.id,  (<Point>featureJson.geometry).coordinates, entity.icon));
+					} else {
+						newEntities.push(this.getPoint(entity.id, (<Point>featureJson.geometry).coordinates, entity.style));
+					}
+					break;
+				}
+				case 'LineString': {
+					newEntities.push(this.getLineString(entity.id, (<LineString>featureJson.geometry).coordinates, entity.style));
+					break;
+				}
+				case 'Polygon': {
+					newEntities.push(this.getPolygon(entity.id, (<Polygon>featureJson.geometry).coordinates, entity.style));
+					break;
+				}
+				case 'MultiPoint': {
+					// Adding each point
+					let i = 0;
+					(<MultiPoint>featureJson.geometry).coordinates.forEach((ptCoords) => {
+						newEntities.push(this.getPoint(`${entity.id}_${i++}`, ptCoords, entity.style));
+					})
+					break;
+				}
+				case 'MultiLineString': {
+					// Adding each line
+					let i = 0;
+					(<MultiLineString>featureJson.geometry).coordinates.forEach((lineCoords) => {
+						newEntities.push(this.getLineString(`${entity.id}_${i++}`, lineCoords, entity.style));
+					})
+					break;
+				}
+				case 'MultiPolygon': {
+					// Adding each poly
+					let i = 0;
+					(<MultiPolygon>featureJson.geometry).coordinates.forEach((polyCoords) => {
+						newEntities.push(this.getPolygon(`${entity.id}_${i++}`, polyCoords, entity.style));
+					})
+					break;
+				}
+				default: {
+					console.warn(`"${featureJson.geometry.type}" Geometry not support`)
+				}
+
 			}
+			// Setting the label
+			if (entity.label && entity.label.text && newEntities.length > 0) {
+				newEntities[0].label = new Cesium.LabelGraphics({
+						text: entity.label.text,
+						font: new Cesium.ConstantProperty( entity.labelSize ? entity.labelSize : undefined),
+					}
+				);
+			}
+
+			// Add new entities to dataSource == Map
+			newEntities.forEach(entity => this.dataSource.entities.add(entity));
```

Comment:
you don't update entities (as the function name implements but duplicates them)

Reply:
Done